### PR TITLE
Improved visualization

### PR DESCRIPTION
**Pull request by pkreissl**
Description of changes:
 - Add visualization of `HollowConicalFrustum` (relies on OpenGL Extrusion library).
 - Improve visualization of `SimplePore` when OpenGL Extrusion library is present.
 - Add capability to handle `Union`s in the visualizer.

The required OpenGL Extrusion library can typically be installed using the standard package manager. It is called `libgle3-dev` on Ubuntu, `libgle` on Fedora.

This PR might depend on issue #3550 &ndash; I implemented the correct visualization, which at the moment does not match what happens in the core ~bur~ but rather what would be expected.

### CODE REVIEW

File: src/python/espressomd/visualization_opengl.pyx [hunk 1]
```diff
@@ -783,19 +789,47 @@ class openGLLive:
...
         coll_shape_obj = collections.defaultdict(list)
-        for c in self.system.constraints:
-            if type(c) == espressomd.constraints.ShapeBasedConstraint:
-                t = c.get_parameter('particle_type')
-                s = c.get_parameter('shape')
-                n = s.name()
-                if n in primitive_shapes:
-                    coll_shape_obj[n].append([s, t])
+        for constraint in self.system.constraints:
+            if type(constraint) == espressomd.constraints.ShapeBasedConstraint:
```
Comment (KaiSzuttor) [inline src/python/espressomd/visualization_opengl.pyx]: ```isinstance(constraint, espressomd.constraints.ShapeBasedConstraint)```

File: src/python/espressomd/visualization_opengl.pyx [hunk 2]
```diff
@@ -783,19 +789,47 @@ class openGLLive:
...
+                ptype = constraint.get_parameter('particle_type')
```
Comment (KaiSzuttor) [inline src/python/espressomd/visualization_opengl.pyx:828]: pep8  says: ```particle_type``` or ```p_type``` or ```part_type```

File: src/python/espressomd/visualization_opengl.pyx [hunk 3]
```diff
@@ -783,19 +789,47 @@ class openGLLive:
         # Collect shapes and interaction type (for coloring) from constraints
         primitive_shapes = [
             'Shapes::Wall', 'Shapes::Cylinder', 'Shapes::Ellipsoid',
-            'Shapes::SimplePore', 'Shapes::Slitpore', 'Shapes::Sphere',
-            'Shapes::SpheroCylinder']
+            'Shapes::HollowConicalFrustum', 'Shapes::SimplePore',
+            'Shapes::Slitpore', 'Shapes::Sphere', 'Shapes::SpheroCylinder']
+
+        def get_shapes_from_union(union):
```
Comment (KaiSzuttor) [inline src/python/espressomd/visualization_opengl.pyx]: if you want to hide the special case of the union, you could differentiate between the union and the others within this function. You could also probably get rid of ```update_shapes_collection``` and fail late. Just try drawing the ```HollowConicalFrustum``` with GLE and if you fail fall back to ```Shapes::Misc```.

**Review by KaiSzuttor — CHANGES_REQUESTED** [review]
Looks good, maybe we can get rid of some special handling

File: src/python/espressomd/visualization_opengl.pyx (hunk 2 above)
Comment (pkreissl) [reply src/python/espressomd/visualization_opengl.pyx:828]: As discussed offline, pep8 will be part of a general refactor of the visualizer.

**Review by KaiSzuttor — APPROVED** [review]

**Pull request merged**